qiskit/quantum_info/operators/symplectic/sparse_pauli_op.py

```diff
@@ -388,10 +398,15 @@ def simplify(self, atol=None, rtol=None):
         # Pack bool vectors into np.uint8 vectors by np.packbits
         array = np.packbits(self.paulis.x, axis=1) * 256 + np.packbits(self.paulis.z, axis=1)
         _, indexes, inverses = np.unique(array, return_index=True, return_inverse=True, axis=0)
-        coeffs = np.zeros(indexes.shape[0], dtype=complex)
+        coeffs = np.zeros(indexes.shape[0], dtype=self.coeffs.dtype)
         np.add.at(coeffs, inverses, self.coeffs)
-        # Delete zero coefficient rows
-        is_zero = np.isclose(coeffs, 0, atol=atol, rtol=rtol)
+        # Delete zero coefficient rows (Ignore if dealing with Parameters)
+        is_zero = []
+        for coeff in coeffs:
+            if isinstance(coeff, (Parameter, ParameterExpression)):
+                is_zero.append(False)
+            else:
+                is_zero.append(np.isclose(coeff, 0, atol=atol, rtol=rtol))
```

Comment:
This is going to have a really big impact on performance, even for coefficient arrays that have `dtype=complex`.  We need to maintain all the fastest paths in the case that this new functionality isn't being used.